### PR TITLE
GG-37101 Improve logging when message deserialization fails

### PR DESCRIPTION
**Pull request by rpuch**
https://ggsystems.atlassian.net/browse/GG-37101

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/direct/stream/v2/DirectByteBufferStreamImplV2.java [hunk 1]
```diff
@@ -271,10 +272,10 @@ public class DirectByteBufferStreamImplV2 implements DirectByteBufferStream {
     private int readItems;
 
     /** */
-    private Object[] objArr;
+    Object[] objArr;
```
Comment (tkalkirill) [inline modules/core/src/main/java/org/apache/ignite/internal/direct/stream/v2/DirectByteBufferStreamImplV2.java:275]: I propose to return the private access modifier, use `GridTestUtils#setFieldValue` in tests.

File: modules/core/src/main/java/org/apache/ignite/internal/direct/stream/v2/DirectByteBufferStreamImplV2.java [hunk 2]
```diff
@@ -1803,7 +1804,42 @@ protected Object read(MessageCollectionItemType type, MessageReader reader) {
...
+    }
+
+    /**
+     * Builds a string showing the simple types of the elements of the list; if the list is longer than the
+     * limit, then only its final elements are considered and ellipsis represents the omitted part of the list.
+     *
+     * @param list List to represent.
+     * @return String like [Integer, String] if there are exactly 2 elements of types Integer and String, respectively.
+     */
+    private String lastTypesWithLimit(List<Object> list) {
```
Comment (tkalkirill) [inline modules/core/src/main/java/org/apache/ignite/internal/direct/stream/v2/DirectByteBufferStreamImplV2.java:1820]: Do I understand correctly that this method can be static, and we can specifically test it?

I propose specifically to test this method better and make the limit an argument of the method, as an option, move this method to an auxiliary class.

File: modules/core/src/test/java/org/apache/ignite/internal/direct/stream/v2/DirectByteBufferStreamImplV2Test.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
+/*
+* Copyright 2019 GridGain Systems, Inc. and Contributors.
```
Comment (tkalkirill) [inline modules/core/src/test/java/org/apache/ignite/internal/direct/stream/v2/DirectByteBufferStreamImplV2Test.java:2]:
```suggestion
* Copyright 2023 GridGain Systems, Inc. and Contributors.
```

File: modules/core/src/main/java/org/apache/ignite/internal/direct/stream/v2/DirectByteBufferStreamImplV2.java (hunk 2 above)
Comment (rpuch) [reply modules/core/src/main/java/org/apache/ignite/internal/direct/stream/v2/DirectByteBufferStreamImplV2.java]: Made the method static and package local to make it easier to test it. I don't think it should be extracted to a common tool yet as it's only used in one class for now. If we need it elsewhere, we'll do it.

File: modules/core/src/test/java/org/apache/ignite/testsuites/IgniteMarshallerSelfTestSuite.java [hunk 4]
```diff
@@ -33,6 +34,7 @@
 import org.junit.runner.RunWith;
 import org.junit.runners.Suite;
 
+
```
Comment (tkalkirill) [inline modules/core/src/test/java/org/apache/ignite/testsuites/IgniteMarshallerSelfTestSuite.java:37]:
```suggestion
```

**Review by tkalkirill — APPROVED** [review]

**Pull request merged**